### PR TITLE
Remove unused option core

### PR DESCRIPTION
**Pull request by mfmweerdesteijn**
Pull Request Checklist. Please read and check each box with an X. Delete any part not applicable. Ask on the [forum](https://community.geodynamics.org/c/aspect) if you need help with any step.

*Describe what you did in this PR and why you did it.*

### Before your first pull request:

* [x] I have read the guidelines in our [CONTRIBUTING.md](../blob/main/CONTRIBUTING.md) document.

### For all pull requests:

* [x] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

### For new features/models or changes of existing features:

* [ ] I have tested my new feature locally to ensure it is correct.
* [ ] I have [created a testcase](http://www.math.clemson.edu/~heister/manual.pdf#sec%3Awriting_tests) for the new feature/benchmark in the [tests/](../blob/main/tests/) directory.
* [ ] I have added a changelog entry in the [doc/modules/changes](../blob/main/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

File: source/simulator/core.cc [hunk 1]
```diff
@@ -109,12 +109,8 @@ namespace aspect
       if (geometry_model.has_curved_elements())
         return std::make_unique<MappingQCache<dim>>(4);
 
-#if !DEAL_II_VERSION_GTE(9,4,0) || DEAL_II_VERSION_GTE(9,4,1)
```
Comment (bangerth) [inline source/simulator/core.cc:112]: Just talking myself through this: The condition means that deal.II either (i) is prior to 9.4, or (ii) 9.4.1 or later. The former cannot happen any more, but it is conceivable that someone uses deal.II 9.4.0. So I think the right change here is to replace this line by
```
#if DEAL_II_VERSION_GTE(9,4,1)
```
and leave the rest of the code we have here.

**Review by bangerth — CHANGES_REQUESTED** [review]

File: source/simulator/core.cc (hunk 1 above)
Comment (mfmweerdesteijn) [reply source/simulator/core.cc:112]: Oh right, looked over the '!'

**Review by tjhei — APPROVED** [review]

**Review by bangerth — APPROVED** [review]
Yes, looks good now!

**Comment by bangerth** [comment]
Do you know how to "squash" the six commits into one? This way, there would only be one commit that encapsulates everything, and the various detours you took will no longer be visible :-)

**Review by bangerth — APPROVED** [review]

**Pull request merged**